### PR TITLE
refactor(pubsub): move get topic, update topic, and list topic samples

### PR DESCRIPTION
**Pull request by alevenberg**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13527)

### CODE REVIEW

File: google/cloud/pubsub/samples/samples.cc [hunk 1]
```diff
@@ -145,61 +145,6 @@ void CreateTopic(google::cloud::pubsub::TopicAdminClient client,
...
-    std::cout << "The topic was successfully updated: " << topic->DebugString()
-              << "\n";
-  }
-  //! [update-topic]
-  (std::move(client), argv.at(0), argv.at(1));
-}
-
-void ListTopics(google::cloud::pubsub::TopicAdminClient client,
-                std::vector<std::string> const& argv) {
-  //! [START pubsub_list_topics] [list-topics]
```
[the remark this answers is not in the document]
Comment (alevenberg) [reply google/cloud/pubsub/samples/samples.cc:186]: Ack

File: google/cloud/pubsub/topic_admin_client.h [hunk 2]
```diff
@@ -139,9 +139,6 @@ class TopicAdminClient {
    *
    * @par Idempotency
    * This is a read-only operation and therefore always idempotent and retried.
-   *
-   * @par Example
-   * @snippet samples.cc list-topics
```
Comment (dbolduc) [inline google/cloud/pubsub/topic_admin_client.h:144]: why not point to `topic_admin_samples.cc`?

File: google/cloud/pubsub/topic_admin_client.h [hunk 3]
```diff
@@ -228,9 +225,6 @@ class TopicAdminClient {
    * @par Idempotency
    * This is a read-only operation and therefore always idempotent and retried.
    *
-   * @par Example
-   * @snippet samples.cc list-topic-snapshots
```
Comment (dbolduc) [inline google/cloud/pubsub/topic_admin_client.h:232]: same

Comment (alevenberg) [reply google/cloud/pubsub/topic_admin_client.h:232]: Because this is the handwritten API. It should just be deprecated.

Comment (dbolduc) [reply google/cloud/pubsub/topic_admin_client.h:232]: That is reasonable. The sample with the generated client might show customers how to update their code. I don't care either way. But whatever you decide to do should be consistent:

https://github.com/googleapis/google-cloud-cpp/blob/4ab32bf4d4eb6c9b0870ff45493d676352a43a49/google/cloud/pubsub/topic_admin_client.h#L108

^ here we reference the sample with the generated client.

Comment (alevenberg) [reply google/cloud/pubsub/topic_admin_client.h:232]: Yeah that's fair. 

removed.

**Review by dbolduc — APPROVED** [review]
Approved, but hold off on merging until release happenings subside

**Comment by alevenberg** [comment]
```Step #3: //google/cloud/workstations:v1_samples_workstations_client_samples (cached) PASSED in 2.8s
Step #3: //google/cloud/batch:v1_samples_samples                                  FAILED in 3 out of 3 in 21.4s
Step #3:   Stats over 3 runs: max = 21.4s, min = 4.6s, avg = 15.8s, dev = 7.9s
Step #3:   /h/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/google_cloud_cpp/bazel-out/k8-fastbuild/testlogs/google/cloud/batch/v1_samples_samples/test.log
Step #3:   /h/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/google_cloud_cpp/bazel-out/k8-fastbuild/testlogs/google/cloud/batch/v1_samples_samples/test_attempts/attempt_1.log
Step #3:   /h/.cache/bazel/_bazel_root/eab0d61a99b6696edb3d2aff87b585e8/execroot/google_cloud_cpp/bazel-out/k8-fastbuild/testlogs/google/cloud/batch/v1_samples_samples/test_attempts/attempt_2.log
Step #3: 
Step #3: Executed 1 out of 662 tests: 661 tests pass and 1 fails locally.
Step #3: There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.```

Seems unrelated...

**Pull request merged**